### PR TITLE
CORE-72342 XDM Schema for Hygiene Operations

### PR DESCRIPTION
**Pull request by anandphatak**
Please link to the issue #…

### CODE REVIEW

File: schemas/hygiene/aep-hygiene-ops-record.example.1.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "xdm:batchTime": "2022-03-03T18:57:00+00:00",
+  "xdm:operationType": "identityDelete",
```
Comment (perlmonger42) [inline schemas/hygiene/aep-hygiene-ops-record.example.1.json:3]: Please make this `"xdm:action": "delete_identity"`.

File: schemas/hygiene/aep-hygiene-ops-record.example.1.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "xdm:targetDatasetId": "1234567890ABCDEFGH",
+  "xdm:operation": {
+    "xdm:identityDelete": {
```
Comment (perlmonger42) [inline schemas/hygiene/aep-hygiene-ops-record.example.1.json:6]: Please make this "xdm:delete_identity".

File: schemas/hygiene/aep-hygiene-ops-record.example.2.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "xdm:operationType": "fieldUpdate",
```
Comment (perlmonger42) [inline schemas/hygiene/aep-hygiene-ops-record.example.2.json:2]: Please make this "update_field".

File: schemas/hygiene/aep-hygiene-ops-record.example.2.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  "xdm:targetDatasetId": "1234567890ABCDEFGH",
+  "xdm:operation": {
+    "xdm:fieldUpdate": {
```
Comment (perlmonger42) [inline schemas/hygiene/aep-hygiene-ops-record.example.2.json:5]: Please make this "xdm:update_field".

File: schemas/hygiene/aep-hygiene-ops-record.schema.json [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  "meta:license": [
+    "Copyright 2022 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/experience/aep-hygiene-ops",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "AEP Hygiene Operation Request",
+  "description": "AEP Hygiene operation request that instructs AEP to deletes or modify records in the specified dataset and sandbox",
```
Comment (perlmonger42) [inline schemas/hygiene/aep-hygiene-ops-record.schema.json:11]: "AEP Data Hygiene operations that instruct AEP to delete or modify records in a specified dataset and sandbox"

File: schemas/hygiene/aep-hygiene-ops-record.schema.json [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  "definitions": {
+    "hygieneOps": {
+      "properties": {
+        "xdm:batchTime": {
+          "title": "Time the batch was created",
+          "description": "Time the hygiene batch was created",
+          "type": "string",
+          "format": "date-time"
+        },
+        "xdm:operationType": {
```
Comment (perlmonger42) [inline schemas/hygiene/aep-hygiene-ops-record.schema.json:25]: Please make this "xdm:action".

File: schemas/hygiene/aep-hygiene-ops-record.schema.json [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+          "title": "Name of the hygiene operation to be applied by downstream consumers",
+          "description": "Name of the hygiene operation to be applied by downstream consumers.",
+          "type": "string",
+          "enum": ["identityDelete", "fieldUpdate"]
```
Comment (perlmonger42) [inline schemas/hygiene/aep-hygiene-ops-record.schema.json:29]: Please make this `["delete_identity", "update_field"]`.

File: schemas/hygiene/aep-hygiene-ops-record.schema.json [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        },
+        "xdm:targetDatasetId": {
+          "title": "Target Dataset ID specified by the user to apply hygiene operations",
+          "description": "May be \"*\", but only if operationType = \"identityDelete\". Otherwise a valid value is required.",
```
Comment (perlmonger42) [inline schemas/hygiene/aep-hygiene-ops-record.schema.json:33]: The "identityDelete" part of this should be changed to "delete_identity".

File: schemas/hygiene/aep-hygiene-ops-record.schema.json [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+          "type": "string"
+        },
+        "xdm:operation": {
+          "title": "Payload of the hygiene op request",
+          "description": "Contents of this payload provides identities to delete, or records to update",
+          "type": "object",
+          "properties": {
+            "xdm:identityDelete": {
```
Comment (perlmonger42) [inline schemas/hygiene/aep-hygiene-ops-record.schema.json:41]: Please make this "xdm:delete_identity", so the field name matches the action value.

File: schemas/hygiene/aep-hygiene-ops-record.schema.json [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+                        "title": "Experience identifier",
+                        "type": "string",
+                        "description": "Experience identifier"
+                      }
+                    }
+                  }
+                }
+              }
+            },
+            "xdm:fieldUpdate": {
```
Comment (perlmonger42) [inline schemas/hygiene/aep-hygiene-ops-record.schema.json:69]: Please make this "update_field", so the field name matches the action value.

**Review by perlmonger42 — CHANGES_REQUESTED** [review]
The structure of this looks right.

It is unfortunate that Siphon needs the field update data be a JSON string, but we can live with it.

I am asking for a few changes, to reflect these decisions the Privacy team has made:
1. We want to use "action" rather than "operationType" as the name of the operator field.
2. We want to use the "verb_noun" style for operator values rather than "nounVerb". I.e., instead of "identityDelete" and "fieldUpdate", we want to use "delete_identity" and "update_field".

I've added a comment at each line where these changes affect the XDM.

File: schemas/hygiene/aep-hygiene-ops-record.example.1.json (hunk 1 above)
Comment (anandphatak) [reply schemas/hygiene/aep-hygiene-ops-record.example.1.json:3]: Done

File: schemas/hygiene/aep-hygiene-ops-record.example.1.json (hunk 2 above)
Comment (anandphatak) [reply schemas/hygiene/aep-hygiene-ops-record.example.1.json:6]: Done

File: schemas/hygiene/aep-hygiene-ops-record.example.2.json (hunk 3 above)
Comment (anandphatak) [reply schemas/hygiene/aep-hygiene-ops-record.example.2.json:2]: Done

File: schemas/hygiene/aep-hygiene-ops-record.example.2.json (hunk 4 above)
Comment (anandphatak) [reply schemas/hygiene/aep-hygiene-ops-record.example.2.json:5]: Done

File: schemas/hygiene/aep-hygiene-ops-record.schema.json (hunk 5 above)
Comment (anandphatak) [reply schemas/hygiene/aep-hygiene-ops-record.schema.json:11]: Done

File: schemas/hygiene/aep-hygiene-ops-record.schema.json (hunk 6 above)
Comment (anandphatak) [reply schemas/hygiene/aep-hygiene-ops-record.schema.json:25]: Done

File: schemas/hygiene/aep-hygiene-ops-record.schema.json (hunk 7 above)
Comment (anandphatak) [reply schemas/hygiene/aep-hygiene-ops-record.schema.json:29]: Done

File: schemas/hygiene/aep-hygiene-ops-record.schema.json (hunk 8 above)
Comment (anandphatak) [reply schemas/hygiene/aep-hygiene-ops-record.schema.json:33]: Done

File: schemas/hygiene/aep-hygiene-ops-record.schema.json (hunk 9 above)
Comment (anandphatak) [reply schemas/hygiene/aep-hygiene-ops-record.schema.json:41]: Done

File: schemas/hygiene/aep-hygiene-ops-record.schema.json (hunk 10 above)
Comment (anandphatak) [reply schemas/hygiene/aep-hygiene-ops-record.schema.json:69]: Done

**Review by perlmonger42 — APPROVED** [review]
LGTM

**Review by prabhum2 — APPROVED** [review]

File: schemas/hygiene/aep-hygiene-ops-record.schema.json [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+                        "title": "Experience identifier",
+                        "type": "string",
+                        "description": "Experience identifier"
+                      }
+                    }
+                  }
+                }
+              }
+            },
+            "xdm:update_field": {
```
Comment (mjasrotia) [inline schemas/hygiene/aep-hygiene-ops-record.schema.json:69]: Please rename it to "updateField" as we do not allow/recommend "_" or "-" in propertynames

File: schemas/hygiene/aep-hygiene-ops-record.schema.json [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+          "type": "string",
+          "format": "date-time"
+        },
+        "xdm:action": {
+          "title": "Name of the hygiene operation to be applied by downstream consumers",
+          "description": "Name of the hygiene operation to be applied by downstream consumers.",
+          "type": "string",
+          "enum": ["delete_identity", "update_field"]
+        },
+        "xdm:targetDatasetId": {
```
Comment (mjasrotia) [inline schemas/hygiene/aep-hygiene-ops-record.schema.json:31]: Please rename to targetDatasetID

**Review by mjasrotia — APPROVED** [review]

**Pull request merged**